db/version_builder.cc

```diff
@@ -32,26 +32,22 @@
 #include "port/port.h"
 #include "table/table_reader.h"
 #include "util/string_util.h"
+#include "version_edit.h"
 
 namespace ROCKSDB_NAMESPACE {
 
 class VersionBuilder::Rep {
-  class NewestFirstBySeqNo {
+  class NewestFirstByL0EpochNumber {
    public:
     bool operator()(const FileMetaData* lhs, const FileMetaData* rhs) const {
       assert(lhs);
       assert(rhs);
 
-      if (lhs->fd.largest_seqno != rhs->fd.largest_seqno) {
-        return lhs->fd.largest_seqno > rhs->fd.largest_seqno;
-      }
-
-      if (lhs->fd.smallest_seqno != rhs->fd.smallest_seqno) {
-        return lhs->fd.smallest_seqno > rhs->fd.smallest_seqno;
+      if (lhs->l0_epoch_number != rhs->l0_epoch_number) {
+        return lhs->l0_epoch_number > rhs->l0_epoch_number;
+      } else {
+        return false;
```

Comment:
[Todo] Replace this as there can be legit echo_number tie due to multiple compaction output files